docs(pathops): extended readme (#55)
canonical/charmtech-charmlibs: _docs/index.md

reference/index
```

This is the Charm Tech team at Canonical's documentation website for charm libraries. It hosts the documentation for the team's charm libraries that are distributed as Python packages, as well as general documentation about charm libraries.

You can also read our {ref}`guidance on distributing charm libraries as Python packages <how-to-python-package>`.

If you're new to charms, see {ref}`Juju | Charm <juju:charm>`.

## Pathops

> Pathops is a Python package providing a {doc}`pathlib <python:library/pathlib>`-like interface for both local filesystem and Kubernetes workload container paths. Charms can use [ContainerPath](pathops.ContainerPath) to interact with files in the workload container, or [LocalPath](pathops.LocalPath) to interact with local files using the same API. Code designed to work for both cases can use [PathProtocol](pathops.PathProtocol) in type annotations.

Reviewer: Very minor - I'd probably say "that provides" instead of providing:

```suggestion
Pathops is a Python package that provides a {doc}`pathlib <python:library/pathlib>`-like interface for both local filesystem and Kubernetes workload container paths. Charms can use [ContainerPath](pathops.ContainerPath) to interact with files in the workload container, or [LocalPath](pathops.LocalPath) to interact with local files using the same API. Code designed to work for both cases can use [PathProtocol](pathops.PathProtocol) in type annotations.
```

> Pathops is [available on PyPI](https://pypi.org/project/charmlibs-pathops). Read the [Pathops reference documentation](pathops).

Reviewer: How about adding a reference to the README:

```suggestion
Pathops is [available on PyPI](https://pypi.org/project/charmlibs-pathops). Read the [Pathops reference documentation](pathops) and [examples of how to get started](https://github.com/canonical/charmtech-charmlibs/blob/main/pathops/README.md#getting-started).
```

Author: I was wondering if there was a nice way to integrate the readme content into the docs site. Do you know any? Presumably I could include the full readme file somewhere as a page if that would be a good idea.

Reviewer: I'm not aware of a way. I haven't looked in great detail though. Maybe a good approach would be to create a how-to guide (later) based on similar examples. Called something like "How to use pathops to write configuration files"

Reviewer 2: In theory you can use literal-include, like:

```
.. literalinclude:: readme.md
        :language: markdown
        :start-at: foo:
        :end-before: bar:
```

However, (a) I don't know if you can do this in myst-markdown or only in ReST, and (b) I'm not sure if you can reference files that are outside of the root docs directory.

*But*, I don't think it's great to do this anyway. A little repetition or a link is fine, I think, and longer term I would expect that there would be more detailed examples in how-to/reference content with the README only having minimal/short ones (because the purpose of the docs are different).

Author: Good point @tonyandrewmeyer , and thanks for the info. I'm thinking I'll go with this:
```suggestion
Pathops is [available on PyPI](https://pypi.org/project/charmlibs-pathops) with examples of how to get started. You can also read the [full reference documentation](pathops).
```
